httpobs/website/api_v2.py

```diff
@@ -0,0 +1,134 @@
+import sys
+from datetime import datetime, timedelta
+
+from flask import Blueprint, jsonify, request
+
+import httpobs.database as database
+import httpobs.scanner as scanner
+from httpobs import STATE_FAILED
+from httpobs.conf import API_COOLDOWN, DEVELOPMENT_MODE
+from httpobs.scanner.grader import get_score_description
+from httpobs.website import add_response_headers
+from httpobs.website.utils import valid_hostname
+
+api_v2 = Blueprint("api_v2", __name__)
+
+
+@api_v2.route("/analyze", methods=["GET", "OPTIONS", "POST"])
+@add_response_headers(cors=True)
+def api_post_scan_hostname():
+    status_code = 200
+    scan = {}
+    tests = {}
+
+    host = request.args.get("host", "").lower().strip()
+    try:
+        site_id = database.select_site_id(host, create=False)
+    except IOError:
+        return {
+            "error": "database-down",
+            "text": "Unable to connect to database",
+        }, 500
+
+    if site_id is not None:
+        hostname = host
+    else:
+        ip = True if valid_hostname(host) is None else False
+        if ip:
+            return {
+                "error": "invalid-hostname-ip",
+                "text": "Cannot scan IP addresses",
+            }, 400
+
+        hostname = valid_hostname(host) or (
+            valid_hostname("www." + host) if host else False
+        )  # prepend www. if necessary
+        if not hostname:
+            return {
+                "error": "invalid-hostname",
+                "text": f"{host} is an invalid hostname",
+            }, 400
+
+    site_id: int = database.select_site_id(host, create=True)
+    scan = database.select_scan_most_recent_scan(site_id)
+
+    if scan and request.method == "POST":
+        time_since_scan = datetime.now() - scan["end_time"]
+        if time_since_scan < timedelta(seconds=API_COOLDOWN):
+            status_code = 429
```

Comment:
Shouldn't we return an error response immediately here?

Reply:
I'll add comment to explain my intentions: if a POST is too soon, it just behaves as a GET would, returning the most recent scan from the db